### PR TITLE
feat(agent,mcp): per-peer diagnostics surface (getConnectionsReturnsForPeer)

### PR DESCRIPTION
**Pull request by branarakic**
## Summary

First of 5 PRs taking ownership of the soak-postmortem follow-ups (Miles↔Lex 6h bidirectional test, May 16). Adds a per-peer diagnostic surface so we can triage the Window D class of asymmetric reachability bugs without grepping `daemon.log`.

- `DKGAgent.getPeerDiagnostics(peerId)` aggregates the libp2p observability state for one peer.
- **Key field**: `getConnectionsReturnsForPeer` exposes what `libp2p.getConnections(pid)` returns (the peerId-keyed lookup `PeerResolver` uses at step 1), alongside `rawConnectionCount` from walking + filtering all connections. When the two diverge on an otherwise-open peer, libp2p's keyed lookup is filtering out connections the raw walk can see — the smoking-gun signature for the Window D bug (inbound circuit-relay open, outbound `dialProtocol` fails with "no valid addresses for peer").
- Also surfaces per-connection direction + transport + `limited` flag + streams + age, peerStore introspection, per-peer outbox state, and the existing PeerHealth snapshot.
- Surfaced via `GET /api/peer-info` (legacy flat shape preserved for Node UI back-compat) and a new MCP tool `dkg_peer_info`.

This is PR 1 of 5 in the [`libp2p-soak-fixes-ownership` plan](../tree/main). Diagnostic surface ships early so PRs 4 (peerStore reverse-path enrichment) and 5 (`dialProtocol` reuses inbound circuit) can assert against it instead of grepping `daemon.log`.

## Postmortem context

Lex's daemon log during the Window D recovery (`03:47:26Z → 03:51:30Z`):
```
31 connection:open events from cZyjAygv  → ALL dir=inbound
 0 connection:open events outbound until 03:51:29Z
20+ "Flushing 1 pending chat outbox entry/entries for cZyjAygv on reconnect"
 1 successful redelivery at 03:51:30Z (last inbound ~1s before)
```

Opportunistic-flush IS running on inbound (Lex confirmed the code path). The thing that failed was `libp2p.dialProtocol(milesPeerId, ...)` — error: "The dial request has no valid addresses for peer". `libp2p.getConnections(milesPeerId)` was returning 0 even with 31 fresh inbound conns from him on file. That divergence is what `getConnectionsReturnsForPeer` makes visible at-a-glance.

## Defensive posture

Every libp2p call inside `getPeerDiagnostics` is wrapped in try/catch so the route never 500s — this surface is most useful when the network is broken; it must not itself break. peerStore-miss degrades to `peerStore: null` (which IS the diagnostic signal); invalid peerId returns a fully-empty snapshot rather than a 500.

## Test plan

- [x] `packages/agent/test/dkg-agent-diagnostics.test.ts` — 12 unit cases binding the prototype method to a stubbed libp2p. Pins the `rawConnectionCount` ≠ `getConnectionsReturnsForPeer` divergence path that PR 5's repro will assert against. Also covers cold-cache peerStore-miss, invalid peerId, libp2p throws.
- [x] `packages/mcp-dkg/test/health-tools.test.ts` — 5 cases covering the new tool's wiring, schema validation, payload rendering, and Window-D-divergence preservation in the rendered output.
- [x] Existing `drop-sweep.test.ts` locked-count bumped 24 → 25.
- [x] Existing `setup-publish-health.test.ts` smoke check updated to also assert `dkg_peer_info` registers.
- [x] `npx tsc --noEmit` clean on `packages/agent`, `packages/cli`, `packages/mcp-dkg`.
- [x] End-to-end probed against Miles's running daemon — route returns 200, fields populate correctly (full new shape lands after daemon restart since the running dist predates this PR).

## Operational note from the same investigation

Miles's MCP `dkg_send_message` was returning a hard-fail message ("NOT delivered") while the daemon was actually queueing the message (`Queued chat outbox retry #1` log lines confirmed). Investigation found this is **not a code bug** — the MCP server process started Thu 11PM May 14, before the MessageOutbox PR merged Fri 09:45, so it's running pre-`queued`-handling code. The current `chat.ts` has correct `queued` handling and the regression test (`chat-tools.test.ts:127`) already pins it. Operational fix: restart Cursor (which restarts the MCP server). No code change needed.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: packages/mcp-dkg/src/client.ts [hunk 1]
```diff
@@ -49,6 +49,72 @@ export interface DkgClientOptions {
...
+    alive: boolean;
+    latencyMs: number | null;
+    lastSeen: number | null;
+    lastChecked: number;
+  } | null;
+  // Legacy flat fields kept for back-compat.
+  connectionCount: number;
+  transports: Array<'direct' | 'relayed'>;
+  directions: Array<'inbound' | 'outbound'>;
+  remoteAddrs: string[];
```
Comment (branarakic) [inline packages/mcp-dkg/src/client.ts]: `remoteAddrs` is typed as `string[]`, but the daemon now returns `diag.connections.map((c) => c.remoteAddr)`, and `remoteAddr` is intentionally `string | null` when libp2p does not expose a multiaddr. That leaves MCP callers with a false type contract: TypeScript consumers can treat every entry as a string even though the runtime JSON can contain `null`. Please widen this to `Array<string | null>` and add a small client/tool test that exercises the null case.
